### ISSUE TITLE
Accuracy Drop Across TensorFlow Versions When Using Keras 3 Instead of Keras 2

### ISSUE THREAD

**Issue by arthurflor23** [tensorflow/tensorflow#71829]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.15.1, 2.16.2, 2.17.0

### Custom code

Yes

### OS platform and distribution

Linux

### Python version

3.11

### Current behavior?

As mentioned [here](https://github.com/tensorflow/tensorflow/issues/63362#issuecomment-2226996257), a drop in model results was observed in the new version of TensorFlow using the new Keras.

I'm using a CNN+BLSTM+CTC for handwriting recognition tests, and only `tf.keras.layers`, `tf.nn.ctc_loss`, and `tf.nn.ctc_beam_search_decoder` as the API base. So I expanded the tests and indeed observed the following results.

As a comment, the loss didn't reach the minimum during training with the new keras. So I also tried to find the root of the problem in default parameter differences between keras and tf.keras, but without success.

### Standalone code to reproduce the issue

```shell
2.15.1                          (CER ~4.6 and WER ~12.5) -- no messages
2.16.2                          (CER ~5.7 and WER ~15.4) -- PTX messages
2.16.2    + TF_USE_LEGACY_KERAS (CER ~4.7 and WER ~12.0) -- PTX messages
2.17.0rc0                       (CER ~5.8 and WER ~15.6) -- numa and gpu_timer messages
2.17.0rc0 + TF_USE_LEGACY_KERAS (CER ~4.8 and WER ~13.0) -- numa and gpu_timer messages
2.17.0rc1                       (CER ~5.2 and WER ~14.8) -- numa and gpu_timer messages
2.17.0rc1 + TF_USE_LEGACY_KERAS (CER ~4.8 and WER ~13.2) -- numa and gpu_timer messages
2.17.0                          (CER ~5.4 and WER ~16.0) -- numa and gpu_timer messages
2.17.0    + TF_USE_LEGACY_KERAS (CER ~4.6 and WER ~12.6) -- numa and gpu_timer messages
```
```


### Relevant log output

_No response_

**Comment by cuixue** [comment]
您好，邮件已经收到，我会尽快处理的。谢谢

**Comment by mihaimaruseac** [comment]
Should this be opened against Keras repo?
